### DISCUSSION TITLE
how to get rendered event endpoint?

### DISCUSSION THREAD

**Discussion by wbh3110** [nhn/tui.calendar#1321] [Q&A] [answered]
Hi. 
I'm try to developing with vue.js 2.x

Using custom detail popup. 
but I can't find to get rendered event position. 
is there any way to find it?

![detailPopupPosition](https://user-images.githubusercontent.com/62129601/204700150-3b979395-a73f-4c57-825f-425dc3be901b.png)

**Comment by adhrinae** [comment, marked as answer]
@wbh3110 
You can get the DOM element of the rendered event by [`getElement`](https://github.com/nhn/tui.calendar/blob/main/docs/ko/apis/calendar.md#getelement) method.

Then you can leverage [`getBoundingClientRect`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect) method from the element.

**Reply by wbh3110** [reply]
![point](https://user-images.githubusercontent.com/62129601/205233634-1eae4055-5f65-49da-8c34-b0a1621f9088.png)

Thank you for your help. 

However, there is one problem. 
The value called with the 'getElement' method always gets only the first element.
For example, in the situation shown in the picture (events 1 and 2 are the same Event),
I think the method should have a different return value.

By the way, when using a custom popup, can it appear in the same position as using the default popup?
It is difficult to taking a Custom Popup position than expected.

**Reply by adhrinae** [reply]
That is an unexpected bug 😢 . need to fix it. As workaround, you can call the `querySelectorAll` on your own. [the `getElement` method does the same thing with `querySelector`.](https://github.com/nhn/tui.calendar/blob/main/apps/calendar/src/factory/calendarCore.tsx#L716-L718
)

About the position of the popup, you can check out [the source code](https://github.com/nhn/tui.calendar/blob/main/apps/calendar/src/components/popup/eventDetailPopup.tsx#L96-L110). But, I personally recommend to check out the well-defined library like [Floating UI](https://floating-ui.com/).

**Reply by wbh3110** [reply]
OK. Thank you for your support. 😀
